### ISSUE TITLE
Regression v0.18.0: CDKTF Typescript Build Issue for External Module

### ISSUE THREAD

**Issue by sabinayakc** [hashicorp/terraform-cdk#3133]
### Expected Behavior

CDKTF Typescript should compile when using external module. 

```
    {
      "name": "eks-module-irsa",
      "source": "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
    }
```

### Actual Behavior

<img width="815" alt="image" src="https://github.com/hashicorp/terraform-cdk/assets/13154166/4b79c04c-6013-4dbb-89e0-425c13cec494">

### Steps to Reproduce

https://github.com/hashicorp/terraform-cdk/issues/2983

### Versions

language: typescript
cdktf-cli: 0.18.0
node: v16.20.1
cdktf: 0.18.0
constructs: 10.2.70
jsii: null
terraform: 1.4.0
arch: x64
os: linux 5.15.90.1-microsoft-standard-WSL2

### Anything Else?

This might be a regression. Checkout References Below.

### References

- https://github.com/hashicorp/terraform-cdk/pull/2986
- #615

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jsteinich** [comment]
Likely the same as #3128

**Comment by ansgarm** [comment]
Hi @sabinayakc 👋

We just released [0.18.1](https://github.com/hashicorp/terraform-cdk/releases/tag/v0.18.1) which includes the #3131 fix that should solve this problem!
